training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,63 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.IdMismatchException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.utils.EndPoints;
 
+import java.util.List;
 
-@Controller
+@RestController
+@RequestMapping(EndPoints.BOOK_BASE_PATH)
 public class BookController {
+
+    @Autowired
+    BookRepository bookRepository;
+
+
     @GetMapping("/greeting")
-    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "World") String name, Model model) {
+    public String greeting(@RequestParam(name = "name", required = false, defaultValue = "World") String name,
+                           Model model) {
         model.addAttribute("name", name);
         return "greeting";
```

Comment:
Just as a comment, this endpoint will stop working when you replace @Controller with @RestController, it will read "greeting" when you call the endpoint. If you want it to keep working (and also for neatness purposes) this endpoint should go in a different controller (with the @Controller annotation) named "GreetingController", for example. 
For the purposes of this training you can simply remove it 😄

Reply:
You're right! I have to move it to it's own controller. Thank you! 😄